### PR TITLE
[USRMGR] Replace old (leftover) debug MessageBox with Wine macros

### PR DESCRIPTION
**Pull request by KRosUser**
## Purpose

- Current code uses "MessageBox" with leftover of old debug code
- Proposed modification is to use DPRINT instead

JIRA issue: [CORE-18276](https://jira.reactos.org/browse/CORE-18276)

### CODE REVIEW

File: dll/cpl/usrmgr/groupprops.c
Comment (SergeGautherie) [inline dll/cpl/usrmgr/groupprops.c:178]: Alignment issue : don't use tabs.

File: dll/cpl/usrmgr/groupprops.c
Comment (SergeGautherie) [inline dll/cpl/usrmgr/groupprops.c:13]:
```suggestion
#include <wine/debug.h>
#include <wine/unicode.h>
```

And (why) is `unicode.h` actually needed?

File: dll/cpl/usrmgr/groupprops.c
Comment (TAN-Gaming) [inline dll/cpl/usrmgr/groupprops.c:15]: You can put them in `usrmgr.h` once and for all.

File: dll/cpl/usrmgr/groupprops.c
Comment (KRosUser) [reply dll/cpl/usrmgr/groupprops.c:13]: right

File: dll/cpl/usrmgr/groupprops.c
Comment (KRosUser) [reply dll/cpl/usrmgr/groupprops.c:15]: Right

**Review by learn-more — DISMISSED** [review]
Hello,
Thanks for your interest in ReactOS!
We ask our contributors [to use their full name when committing](https://github.com/reactos/reactos/blob/master/CONTRIBUTING.md#rules-and-recommendations).
[Please amend your commit with your full name](https://stackoverflow.com/a/3042512), and update this PR.

To change this for future PR's, you can update it for globally (for the system):
```
git config --global user.name "Your Name"
git config --global user.email you@example.com
```

Or just for the current repository:
```
git config user.name "Your Name"
git config user.email you@example.com
```

Regards,

**Comment by binarymaster** [comment]
Additional details to Mark's request:

1. Please set your full name in [GitHub profile settings](https://github.com/settings/profile) - to make Squash & Merge button correctly setting the name when used by maintainers
1. Make sure to uncheck `Keep my email addresses private` in [GitHub email settings](https://github.com/settings/emails) - for the same Squash & Merge button

File: dll/cpl/usrmgr/usrmgr.h
Comment (HBelusca) [inline dll/cpl/usrmgr/usrmgr.h:5]:
```suggestion
#include <stdarg.h>
#include <wine/debug.h>
```

File: dll/cpl/usrmgr/usrmgr.h
Comment (HBelusca) [inline dll/cpl/usrmgr/usrmgr.h:40]:
```suggestion
```

File: dll/cpl/usrmgr/misc.c
Comment (HBelusca) [inline dll/cpl/usrmgr/misc.c:12]:
```suggestion
```

**Review by HBelusca — COMMENTED** [review]
lgtm after the git name fix.

**Comment by KRosUser** [comment]
@HBelusca @binarymaster : Thanks, all cjhanges done (including name unless i missed something)

**Comment by binarymaster** [comment]
> Thanks, all changes done (including name unless i missed something)

Still no name in GitHub profile, edit the `Name` field here: https://github.com/settings/profile

**Comment by KRosUser** [comment]
> > Thanks, all changes done (including name unless i missed something)
> 
> Still no name in GitHub profile, edit the `Name` field here: https://github.com/settings/profile

Name added

**Comment by learn-more** [comment]
You are missing `WINE_DEFAULT_DEBUG_CHANNEL`

**Comment by KRosUser** [comment]
> You are missing `WINE_DEFAULT_DEBUG_CHANNEL`

Thank you !

**Comment by KRosUser** [comment]
New compiles succesfully. @learn-more : Ready for review !

File: dll/cpl/usrmgr/userprops.c
Comment (learn-more) [inline dll/cpl/usrmgr/userprops.c:479]: Mixing CHAR and TCHAR

**Review by learn-more — CHANGES_REQUESTED** [review]
This won't print correctly

File: dll/cpl/usrmgr/userprops.c
Comment (KRosUser) [reply dll/cpl/usrmgr/userprops.c:479]: Would you recommend wcstombs() here ?

Comment (learn-more) [reply dll/cpl/usrmgr/userprops.c:479]: `szGroupName` can be CHAR or WCHAR, depending on the compiler settings (because it's TCHAR).
That is a very obsolete choice, but sadly some code still uses TCHAR.
So either rewrite the entire thing to use WCHAR, or take into account the current setting and conditionally (compile time!) use the correct format specifier here.

Comment (KRosUser) [reply dll/cpl/usrmgr/userprops.c:479]: Would you agree that the most desirable option is to work on refactoring toward a clean WCHAR implementation for the sake of value for the project ?

File: dll/cpl/usrmgr/usrmgr.h
Comment (HBelusca) [inline dll/cpl/usrmgr/usrmgr.h:6]:
```suggestion
WINE_DEFAULT_DEBUG_CHANNEL(usrmgr);

#ifdef UNICODE
#define dbgstrx(x)      wine_dbgstr_w(x)
#else
#define dbgstrx(x)      (const char*)(x)
#endif
```

File: dll/cpl/usrmgr/groupprops.c
Comment (HBelusca) [inline dll/cpl/usrmgr/groupprops.c:176]:
```suggestion
            TRACE("Selected user: %s", dbgstrx(szUserName));
```

File: dll/cpl/usrmgr/userprops.c
Comment (HBelusca) [inline dll/cpl/usrmgr/userprops.c:479]:
```suggestion
                    TRACE("Selected group: %s", dbgstrx(szGroupName));
```

Comment (HBelusca) [reply dll/cpl/usrmgr/userprops.c:479]: To be left to the maintainer of usrmgr (Eric).

Comment (HBelusca) [reply dll/cpl/usrmgr/userprops.c:479]: To be managed by the maintainer of usrmgr (Eric).

Comment (KRosUser) [reply dll/cpl/usrmgr/userprops.c]: Thanks @HBelusca for the finalization.
OK. So your change is now making the PR OK isn't it ?

**Review by HBelusca — APPROVED** [review]

File: dll/cpl/usrmgr/userprops.c
Comment (KRosUser) [reply dll/cpl/usrmgr/userprops.c]: @learn-more : Do you approve it to, are are still changes required ?

File: dll/cpl/usrmgr/usrmgr.h
Comment (HBelusca) [inline dll/cpl/usrmgr/usrmgr.h:6]:
```suggestion

WINE_DEFAULT_DEBUG_CHANNEL(usrmgr);
```

**Comment by binarymaster** [comment]
@KRosUser you have not completed this step that I mentioned here 6 days ago:
> Make sure to uncheck `Keep my email addresses private` in [GitHub email settings](https://github.com/settings/emails) - for the same Squash & Merge button

So your email in merged commits is not correct right now: https://git.reactos.org/?p=reactos.git;a=commit;h=de5af76811bbd318cdef46e193f306833e3fa740

**Comment by KRosUser** [comment]
> @KRosUser you have not completed this step that I mentioned here 6 days ago:
> 
> > Make sure to uncheck `Keep my email addresses private` in [GitHub email settings](https://github.com/settings/emails) - for the same Squash & Merge button
> 
> So your email in merged commits is not correct right now: https://git.reactos.org/?p=reactos.git;a=commit;h=de5af76811bbd318cdef46e193f306833e3fa740

Fixed

**Pull request merged**
